specs/switch-driver-capabilities.md

```diff
@@ -0,0 +1,51 @@
+
+# Expose network driver capabilites
+-----------
+
+[Issue 410](https://github.com/CCI-MOC/hil/issues/410)
+
+We want to expose switch driver capabilities. Capabilities could include:
+
+- Jumbo frames are per-switch on our Dell switches and per-port on our Ciscos (#755)
+-  QoS (#384)
+- Disabling STP (#779)
+- DHCP/ARP Snooping (#471)
+- Enabling the tagging of DHCP packets with the port number (from @okrieg or @pjd-nu)
+
+# Solution
+-----------
+
+* All switch drivers would have a method called get_capabilities.
+
+* This method would return a list with all capabilities supported by the switch.
+Its implementation could be flexible; it could simply return a hardcoded list, or generate it from some database.
+
+* The superclass will have a method `has_capbility(self, name)` that would return
+`name in self.get_capabilities()`. Drivers can overwrite if need be.
+
+* an API that needs to check for capabilities will call `has_capability()` and
```

Comment:
now that we have `ensure_legal_operation()` in the switches, I think `has_capability()` becomes redundant. capability checks can also be done be in `ensure_legal_operation()`

Reply:
We've already established that get_capabilities is sufficient. We can leave has_capability out and add it later if we want.

Reply:
coo. I got rid of it, and rebased this PR.